forem/forem
# OpenGraph fallback for Embeds (merge-by-any-core)

## What type of PR is this? (check all applicable)


## Description
Using the [MetaInspector](https://github.com/metainspector/metainspector) gem, this PR implements the OpenGraph fallback for Embeds and renders a simple A-tag when the embedded link lacks necessary OG tags.

NOTE: this PR handles only the parsing of OG data. In a future PR, we shall also [incorporate Twitter data](https://github.com/forem/forem/issues/17054).

Design Specs are here: https://www.figma.com/file/fJuPU4FRJQw63virkPiC9f/Liquid-tags?node-id=502%3A53

## Related Tickets & Documents
- closes #16688
- closes #16689
- closes #16690


## QA Instructions, Screenshots, Recordings
- Create an embed of a URL without a LiquidTag (e.g. https://ourtimefortech.org or https://zoom.us). Preview your post; you should see your embedded link rendered with the Page Title and Description, and Image if one exists.
- Create an embed using https://example.com - this URL does not have OG tags. A simple A-tag should render.

### UI accessibility concerns?
None detected; will make suggested improvements if made.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?
